### PR TITLE
feat(hbase): version 2.6.0

### PR DESCRIPTION
**Pull request by razvan**
# Description

Fixes https://github.com/stackabletech/issues/issues/589

Part of: https://github.com/stackabletech/hbase-operator/issues/488

A first image **without** Phoenix and hbase-operator-tools was built with `bake --product hbase=2.6.0 --organization sandbox` and is available [here](https://repo.stackable.tech/#browse/browse:docker:v2%2Fsandbox%2Fhbase%2Ftags%2F2.6.0-stackable0.0.0-dev)

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -246,21 +260,9 @@
...
-                "java-devel": "11"
-            },
-            {
-                "product": "1.25.0",
-                "java-base": "21",
-                "java-devel": "11"
-            },
+            {"product": "1.21.0", "java-base": "11", "java-devel": "11"},
+            {"product": "1.23.2", "java-base": "11", "java-devel": "11"},
+            {"product": "1.25.0", "java-base": "21", "java-devel": "11"},
```
Comment (maltesander) [inline conf.py:265]: What are the reasons for this :) ?

File: hbase/Dockerfile [hunk 2]
```diff
@@ -104,6 +106,21 @@ RUN export LIBS=$(find /stackable/hadoop/share/hadoop -name '*.jar' -printf '%p:
...
+    https://repo.stackable.tech/repository/packages/jackson-dataformat-xml/jackson-dataformat-xml-${JACKSON_DATAFORMAT_XML}.jar; \
+    fi
+
+# ===
+# OPA Authorizer
+RUN if [[ -n "$OPA_AUTHORIZER" ]]; then \
+  git clone --depth 1 --branch "$OPA_AUTHORIZER" https://github.com/stackabletech/hbase-opa-authorizer.git && \
+  mvn -DskipTests package -fhbase-opa-authorizer  && \
+  cp target/hbase-opa-authorizer*.jar /stackable/hbase-${PRODUCT}/lib; \
+  fi
```
Comment (maltesander) [inline hbase/Dockerfile]:
```suggestion
RUN if [[ -n "$JACKSON_DATAFORMAT_XML" ]]; then \
      curl -L --fail -o /stackable/hbase-${PRODUCT}/lib/jackson-dataformat-xml-${JACKSON_DATAFORMAT_XML}.jar \
      https://repo.stackable.tech/repository/packages/jackson-dataformat-xml/jackson-dataformat-xml-${JACKSON_DATAFORMAT_XML}.jar; \
    fi

# ===
# OPA Authorizer
RUN if [[ -n "$OPA_AUTHORIZER" ]]; then \
      git clone --depth 1 --branch "$OPA_AUTHORIZER" https://github.com/stackabletech/hbase-opa-authorizer.git && \
      mvn -DskipTests package -fhbase-opa-authorizer  && \
      cp target/hbase-opa-authorizer*.jar /stackable/hbase-${PRODUCT}/lib; \
    fi
```

File: conf.py (hunk 1 above)
Comment (razvan) [reply conf.py:265]: talk to my editor :)

Comment (maltesander) [reply conf.py:265]: Hello Mr. Editor, what are the reasons for this :) ?

File: hbase/Dockerfile (hunk 2 above)
Comment (razvan) [reply hbase/Dockerfile]: applied a variant of this: https://github.com/stackabletech/docker-images/pull/701/commits/23b3ec317c4a7f101023aa2b46d33530f9e75b33

**Review by sbernauer — DISMISSED** [review]
I ran the current smoke test (from main) on this image 3 times, the `/stackable/phoenix/bin/psql.py /stackable/phoenix/examples/WEB_STAT.sql /stackable/phoenix/examples/WEB_STAT.csv /stackable/phoenix/examples/WEB_STAT_QUERIES.sql` did hang up every time.
This is the same issue we had with HBase 2.5 :(

File: hbase/Dockerfile [hunk 3]
```diff
@@ -31,16 +31,10 @@ USER stackable
 WORKDIR /stackable
 
 COPY --chown=stackable:stackable hbase/stackable/patches /stackable/patches
-COPY --chown=stackable:stackable hbase/stackable/jmx /stackable/jmx
+COPY --chown=stackable:stackable hbase/stackable/jmx/config${JMX_EXPORTER} /stackable/jmx
 
 # Cache mounts are owned by root by default
 # We need to explicitly give the uid to use which is hardcoded to "1000" in stackable-base
-# And every cache needs its own id, we can't share them between stages because we might delete the caches
```
Comment (lfrancke) [inline hbase/Dockerfile:38]: I think we should keep this in

File: hbase/Dockerfile [hunk 4]
```diff
@@ -87,6 +83,40 @@ if [ "${DELETE_CACHES}" = "true" ] ; then
...
+
+RUN --mount=type=cache,id=maven-opa,uid=1000,target=/stackable/.m2/repository <<EOF
+
+###
+### OPA Authorizer (only for 2.6 upwards)
+###
+if [[ -n "$OPA_AUTHORIZER" ]]; then
+    git clone --depth 1 --branch "$OPA_AUTHORIZER" https://github.com/stackabletech/hbase-opa-authorizer.git
+    mvn \
+      --quiet \
```
Comment (lfrancke) [inline hbase/Dockerfile:103]: Can you remove the quiet flag? I like to see at least a bit what's going on

File: hbase/stackable/patches/apply_patches.sh [hunk 5]
```diff
@@ -16,7 +16,8 @@ PATCH_DIR="patches/$VERSION"
 # Check if version-specific patches directory exists
 if [ ! -d "$PATCH_DIR" ]; then
   echo "Patches directory '$PATCH_DIR' does not exist."
-  exit 1
+  # maybe not all versions need patches
+  exit 0
```
Comment (lfrancke) [inline hbase/stackable/patches/apply_patches.sh:20]: I know it's not perfect but in other products we do this by having an empty directory.
Can we do the same here? IMO it's more important to be consistent even if the solution ain't great

File: hbase/versions.py [hunk 6]
```diff
@@ -11,7 +11,9 @@
         "phoenix": "5.2.0",
         "hbase_profile": "2.4",
         "hadoop": "3.3.6",
-        "jmx_exporter": "1.0.1",
+        "jmx_exporter": "1.0.1",  # update the stackable/jmx/config<version> folder too
+        "opa_authorizer": "",  # only for HBase 2.6.0
+        "delete_caches": "true",
```
Comment (lfrancke) [inline hbase/versions.py:16]: I'd still prefer to not have the `delete_caches` stuff in`versions.py`

File: hbase/versions.py [hunk 7]
```diff
@@ -23,6 +25,22 @@
...
+    },
+    {
+        "product": "2.6.0",
+        "hbase_thirdparty": "4.1.7",
+        "hbase_operator_tools": "1.3.0-SNAPSHOT",
```
Comment (lfrancke) [inline hbase/versions.py:35]: We need this to be versioned _somehow_.
For example by referencing a git commit hash

Same for phoenix

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: hbase/stackable/patches/apply_patches.sh (hunk 5 above)
Comment (razvan) [reply hbase/stackable/patches/apply_patches.sh]: Fixed https://github.com/stackabletech/docker-images/pull/701/commits/0afda96d8ab0f590d0c825e23972bab47181bd77

File: hbase/Dockerfile (hunk 4 above)
Comment (razvan) [reply hbase/Dockerfile]: Fixed https://github.com/stackabletech/docker-images/pull/701/commits/0afda96d8ab0f590d0c825e23972bab47181bd77

File: hbase/Dockerfile (hunk 3 above)
Comment (razvan) [reply hbase/Dockerfile:38]: Fixed https://github.com/stackabletech/docker-images/pull/701/commits/0afda96d8ab0f590d0c825e23972bab47181bd77

**Review by lfrancke — APPROVED** [review]

**Pull request merged**